src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -244,7 +244,7 @@ def build_msi_role_assignment(vm_vmss_name, vm_vmss_resource_id, role_definition
 def build_vm_resource(  # pylint: disable=too-many-locals
         cmd, name, location, tags, size, storage_profile, nics, admin_username,
         availability_set_id=None, admin_password=None, ssh_key_value=None, ssh_key_path=None,
-        image_reference=None, os_disk_name=None, custom_image_os_type=None,
+        image_reference=None, os_disk_name=None, custom_image_os_type=None, authentication_type=None,
```

Comment:
@adewaleo, i suggest we don't create a new arg, rather only enable it when a user provides both password and ssh keys.

Reply:
We decided to leave the arg in, after our discussion. Resolving this.